src/clients/dataset-client.ts

```diff
@@ -26,33 +26,35 @@ export class ExtDatasetClient<T extends DatasetItem> extends DatasetClient<T> im
         const { pageSize, ...listItemOptions } = options;
         this.customLogger.info('Iterating Dataset', { pageSize }, { url: this.url });
 
-        let totalItems = 0;
+        const initialOffset = listItemOptions.offset ?? 0;
```

Comment:
nit: I feel weird about this being in the outer scope as it really doesn't matter when we don't have a `pageSize` - that would entail duplicating the log, but TBH I'm not against that as it's actually two implementations masquerading as one function..